proposals/statics.md

```diff
@@ -0,0 +1,1906 @@
+of the corresponding class, making all the static members of the corresponding class available by their short name.
+
+```kotlin
+fun Color.static.parse(s: String): Color { // static extension
+    RED // OK: unqualified `Color.RED` reference to a static member works here
+    this // ERROR: Unresolved: static extension has no real receiver
+    rgb  // ERROR: Unresolved: instance members of the `Color` class are not present in its phantom static receiver
+    // ...
+}
+```
+
+The phantom static implicit receiver is added as a phantom receiver to the receiver chain with the same priority
+as regular receiver for the regular extension function would have been added.
+See the [receivers](https://kotlinlang.org/spec/overload-resolution.html#receivers) section
+in the specification for details.
+
+The reference to static scope `<ClassName>.static` cannot be used by itself as a value or as type.
+See [Static scope projection reference](#static-scope-projection-reference) for
+discussion on potentially allowing it in the future.
+
+Static extensions cannot have `operator` modifier.
+See [Static operators](#static-operators) for potential lifting of this restriction in the future.
+
+### Static extensions on generic classes
+
+Static extensions are defined on a _class_, not on a specific _type_. That means, that when the generic class
+is being extended only the class name is being specified. For example:
+
+```kotlin
+class Box<T>(val value: T)
+
+fun <T> Box.static.of(value: T): Box<T> = Box(value)
+//      ^^^ It is an error to write Box<T> here
+```
+
+### Extensions on static objects
+
+Static extensions can be declared for [Static objects](#static-objects) in exactly the same way
+as [Static extensions](#static-extensions) for classes and interfaces using `<StaticObjectName>.static` as receiver.
+They follow all the same rules as other static extensions.
+
+For example:
+
+```kotlin
+static object Namespace {
+    // static member
+    fun doSomething() {}
+}
+
+// extension on static object
+fun Namespace.static.ext() {
+    this // ERROR: It has no receiver
+    doSometing() // OK: Via phantom static implicit receiver
+}
+```
+
+> Rationale: the requirement to have an explicit `.static` modifier makes it explicit for the reader of this
+> extension function that it has no `this` receiver in scope without having to find the declaration of
+> the object that is being extended to verify that it is a static object. It will also aid with
+> [Migration of objects to static objects](#migration-of-objects-to-static-objects) as the extensions on
+> the objects that are being migrated to static objects can clearly and independently migrate from
+> being regular extensions to static extensions.
+
+### Static extensions resolution and import
+
+Static extensions can be called in several ways. One way is using making a qualified class with
+`<ClassName>.<extensionName>` as in, for example, `Color.parse(s)`. Similarly to the regular extensions, the
```

Comment:
I think this sentence is wrong.
At least I don't understand what is meant by this:
> One way is using making a qualified class with `<ClassName>.<extensionName>` [...]

Reply:
Is it maybe supposed to say "One way is making a qualified call with `<ClassName>.<extensionName>`" ?

Reply:
Yes. That's a typo. class -> call. Fixing it.